### PR TITLE
k8s-infra: verify manifests for registry.k8s.io

### PR DESCRIPTION
**Pull request by ameukam**
Related:
  - https://github.com/kubernetes/k8s.io/issues/3411

Verify the manifests lists registry.k8s.io in replacement of k8s.gcr.io

See: https://groups.google.com/d/msgid/kubernetes-sig-testing/CANw6fcFgKaCLLiuY_8TBp%3DKWROHZbuj1FYWAfyGW6XOQ%3D_nLUA%40mail.gmail.com?utm_medium=email&utm_source=footer

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

**Comment by ameukam** [comment]
cc @dims @neolit123

File: tests/e2e/manifests/verify_manifest_lists.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ fi
 
 # install go if missing
 if ! `go version > /dev/null`; then
-	curl -sSL https://dl.google.com/go/go1.16.linux-amd64.tar.gz -o "/${TMP}/go.tar.gz"
```
Comment (dims) [inline tests/e2e/manifests/verify_manifest_lists.sh:30]: unrelated change @ameukam ?

Comment (ameukam) [reply tests/e2e/manifests/verify_manifest_lists.sh:30]: wanted to bump the version of go in the same time. Will revert.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
Thanks @ameukam
